packages/headless/src/features/folding/folding-state.ts

```diff
@@ -7,6 +7,17 @@ export interface FoldedResult extends Result {
   children: FoldedResult[];
 }
 
+export interface Collection extends FoldedResult {
```

Comment:
Rather than extending, can we remove/rename `FoldedResult`?

Reply:
if we don't expect results to be hierarchical, then we could move the children property into `Collection` and change its type to `Result`. Otherwise, a collection legitimately is a folded result since it contains other folded results. We could remove `extends` and copy the attribute.

We could make it contain a single result instead, but then the folded result list controller could no longer inherit the result list controller.